File: 은지/LV2/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for tree in skill_trees:
        idx = -1 # 가장 최근에 사용된 skill 순서
        chk = 1 # skill 순서 체크
        for j in range(len(tree)):
            for i in range(len(skill)):
                if tree[j] == skill[i]:
                    if i != idx+1: # skill 순서 무시
                        chk = 0
                        break
                    else:
                        idx = i
        if chk == 1:  
            answer += 1
    return answer

File: 은지/LV2/스킬트리.js
function solution(skill, skill_trees) {
  var answer = 0;
  for (let tree of skill_trees) {
    var idx = -1
    var chk = 1
    for (let j=0; j<tree.length; j++) {
      for (let i=0; i<skill.length; i++) {
        if (tree[j] === skill[i]) {
          if (i !== idx + 1) {
            chk = 0
            break
          } else {
            idx = i
          }
        }
      }
    }
    if (chk === 1) {
      answer += 1
    }
  }
  return answer;
}